File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.cc
#include <bits/stdc++.h>
#include <unordered_set>
#include <unordered_map>
using namespace std;

int n, m;
string name;
string a1[100005];
unordered_map<string, int> a2;
string question;

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);

	cin >> n >> m;

	for (int i = 1; i <= n; i++) {
		cin >> a1[i];
		a2[a1[i]] = i;
	}

	while (m--) {
		cin >> question;
		if (isdigit(question[0])) {
			cout << a1[stoi(question)] << "\n";
		}
		else
			cout << a2[question] << "\n";
	}


}

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        HashMap<String, Integer> stoi = new HashMap<>();
        HashMap<Integer, String> itos = new HashMap<>();

        for(int i = 1; i <= N; i++){
            String name = br.readLine();
            stoi.put(name, i);
            itos.put(i, name);
        }

        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < M; i++){
            String question = br.readLine();
            if(Character.isDigit(question.charAt(0))) { // 문자로 시작하는지 확인
                sb.append(itos.get(Integer.parseInt(question))).append("\n");
            } else {
                sb.append(stoi.get(question)).append("\n");
            }
        }

        bw.write(sb.toString());
        bw.flush();
    }
}
